### ISSUE TITLE
Scan git diff for vulnerabilities

### ISSUE THREAD

**Issue by knqyf263** [aquasecurity/trivy#2603]
## Description
Trivy can be integrated into CI easily, but the vulnerability disclosure lifecycle is not synced with the development lifecycle. A job can fail due to newly disclosed vulnerabilities even though nothing has changed in the source code. The vulnerability could affect your production regardless of your job failure if your application is already deployed.

A job can fail on newly disclosed vulnerabilities currently, but it cannot prevent a vulnerable dependency from deploying since it is already in production. It doesn't make sense to stop your deployment.

![image](https://user-images.githubusercontent.com/2253692/181232481-8c329073-3ae2-40d2-a759-691bbf0b0a94.png)

I'd suggest the job should fail on vulnerabilities of newly added dependencies. Then, it can stop deploying the vulnerable dependency. For instance, if there is a change in `package-lock.json`, only the addition will be scanned. Only repository changes, not vulnerability disclosure, will fail CI jobs. 

![image](https://user-images.githubusercontent.com/2253692/181233259-7be6f551-f4bf-4898-a2a3-cb521a11a363.png)

Obviously, people want to know newly disclosed vulnerabilities as well. They should be detected in a cron job, nightly, hourly, etc.

In short, users should have two scanning jobs.

1. Cron job
    - Scan for all vulnerabilities, including newly disclosed one
    - e.g. `$ trivy fs --exit-code 1 ./package-lock.json`
2. CI job
    - Scan only dependency differences for vulnerabilities
    - e.g. `$ trivy fs --exit-code 1 --diff ./package-lock.json`

![image](https://user-images.githubusercontent.com/2253692/181234729-8d54c3ad-5295-4d30-8d09-59695dbbda35.png)

I believe it would be useful, but we welcome any feedback👍

**Comment by phisco** [comment]
+1

**Comment by phisco** [comment]
FYI: [docker scout](https://docs.docker.com/scout/) implements the diff between images: https://github.com/docker/scout-action#compare-to-an-image

**Comment by ngraef** [comment]
This would be nice as a built-in feature. I currently have it implemented as a PR check in a fairly simple pipeline using multiple scans:
1. Checkout base branch in `./base/`
2. Checkout head branch in `./head/`
3. Scan `./base/`, output `base.json`
4. Scan `./head/`, output `head.json`
5. Read the JSON results, calculate sets of findings added and removed by the head branch relative to the base

Because it scans both the base and head branches using the same database version, it achieves the goal of only alerting on findings caused by code changes, not newly disclosed vulnerabilities.

**Comment by saurabhlime** [comment]
How can we get this prioritized?
